linux_os/guide/services/sssd/sssd_enable_pam_services/oval/shared.xml

```diff
@@ -2,28 +2,29 @@
   <definition class="compliance" id="sssd_enable_pam_services" version="1">
     <metadata>
       <title>Configure PAM in SSSD Services</title>
-      <affected family="unix">
-        <platform>Red Hat Enterprise Linux 7</platform>
-        <platform>Red Hat Enterprise Linux 8</platform>
-        <platform>Red Hat Virtualization 4</platform>
-        <platform>multi_platform_ol</platform>
-      </affected>
+      {{{- oval_affected(products) }}}
       <description>SSSD should be configured to run SSSD PAM services.
       </description>
     </metadata>
     <criteria>
-      <criterion comment="check if pam is configured in the services setting of the sssd section"
-      test_ref="test_sssd_enable_pam_services" />
+        <criterion comment="check if pam is configured in the services setting of the sssd section"
+        test_ref="test_sssd_enable_pam_services" />
     </criteria>
   </definition>
-  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  <ind:textfilecontent54_test check="all" check_existence="any_exist"
   comment="check if pam is configured in the services setting of the sssd section"
   id="test_sssd_enable_pam_services" version="1">
     <ind:object object_ref="obj_sssd_enable_pam_services" />
+    <ind:state state_ref="state_sssd_enable_pam_services" />
   </ind:textfilecontent54_test>
+
   <ind:textfilecontent54_object id="obj_sssd_enable_pam_services" version="1">
-    <ind:filepath>/etc/sssd/sssd.conf</ind:filepath>
-    <ind:pattern operation="pattern match">^[\s]*\[sssd]([^\n\[\]]*\n+)+?[\s]*services.*pam.*$</ind:pattern>
+    <ind:filepath operation="pattern match">^/etc/sssd/(sssd|conf\.d/.*)\.conf$</ind:filepath>
+    <ind:pattern operation="pattern match">^[\s]*\[sssd]([^\n\[\]]*\n+)+?[\s]*services[\s]*=[\s]*(.*)$</ind:pattern>
```

Comment:
Another possible issue is that there are two capturing groups in the regex.

Reply:
That might be it.

```
("subexpression" " " ) ("subexpression" "nss,pam" ) ) ) () ) ) [oscap(8378):unknown(7f7fecf0b880):seap-packet.c:903:SEAP_packet_recv]
```

Reply:
I'm going to use https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/services/sssd/sssd_run_as_sssd_user/oval/shared.xml#L23 as a basis for the regex, it seems much more accurate. With one minor change to the capturing group it seems to work just fine.